networkx/convert_matrix.py

```diff
@@ -121,51 +121,101 @@ def to_pandas_dataframe(G, nodelist=None, multigraph_weight=sum, weight='weight'
+    a problem if your row is entirely numeric and a mix of ints and floats. In
+    that case, all values will be returned as floats. See the
+    DataFrame.iterrows documentation for an example.
 
     Parameters
     ----------
     df : Pandas DataFrame
-      An adjacency matrix representation of a graph
+      An edge list representation of a graph
 
-    create_using : NetworkX graph
-       Use specified graph for result.  The default is Graph()
+    source : str or int
+      A valid column name (string or iteger) for the source nodes (for the
+      directed case).
 
-    Notes
-    -----
-    If the numpy matrix has a single data type for each matrix entry it
-    will be converted to an appropriate Python data type.
+    target : str or int
+      A valid column name (string or iteger) for the target nodes (for the
+      directed case).
 
-    If the numpy matrix has a user-specified compound data type the names
-    of the data fields will be used as attribute keys in the resulting
-    NetworkX graph.
+    edge_attr : str or int, iterable, True
+      A valid column name (str or integer) or list of column names that will
+      be used to retrieve items from the row and add them to the graph as edge
+      attributes. If `True`, all of the remaining columns will be added.
+
+    create_using : NetworkX graph
+       Use specified graph for result.  The default is Graph()
 
     See Also
     --------
-    from_pandas_dataframe
+    to_pandas_dataframe
 
     Examples
     --------
     Simple integer weights on edges:
 
     >>> import pandas as pd
-    >>> df=pd.DataFrame([[1,1],[2,1]])
-    >>> G=nx.from_pandas_dataframe(df)
+    >>> import numpy as np
+    >>> r = np.random.RandomState(seed=5)
+    >>> ints = r.random_integers(1, 10, size=(3,2))
+    >>> a = ['A', 'B', 'C']
+    >>> b = ['D', 'A', 'E']
+    >>> df = pd.DataFrame(ints, columns=['weight', 'cost'])
+    >>> df[0] = a
+    >>> df['b'] = b
+    >>> df
+       weight  cost  0  b
+    0       4     7  A  D
+    1       7     1  B  A
+    2      10     9  C  E
+    >>> G=nx.from_pandas_dataframe(df, 0, 'b', ['weight', 'cost'])
+    >>> G['E']['C']['weight']
+    10
+    >>> G['E']['C']['cost']
+    9
     """
 
-    import pandas as pd
-    A = df.values
-    G = from_numpy_matrix(A, create_using)
-    try:
-        df = df[df.index]
-    except:
-        raise nx.NetworkXError("Columns must match Indices.",
-                               "%s not in columns"%list(set(df.index).difference(set(df.columns))))
-    nx.relabel.relabel_nodes(G, dict(enumerate(df.columns)), copy=False)
-    return G
+    g = _prep_create_using(create_using)
+
+    # Index of source and target
+    src_i = df.columns.get_loc(source)
+    tar_i = df.columns.get_loc(target)
+    if edge_attr:
+        # If all additional columns requested, build up a list of tuples
+        # [(name, index),...]
+        if edge_attr == True:
```

Comment:
This works.  Booleans are singletons in Python, so I guess my inclination would be to do:   `if edge_attr is True` in this particular situation---where you are distinguishing an actual boolean from something which evaluates to true.